### PR TITLE
Updates to linters and formatters (now with JS support!)

### PR DESCRIPTION
**Pull request by skearnes**
* Use `clang-format` to clean up *.js and *.proto files
* Use the closure compiler linter to flag style violations in *.js files
* Adjust YAFP targets to include the top-level `ord-schema` directory

Notes:
* `clang-format` and `yapf` now use Google style (improves dict formatting, etc.)
* The JS linter (`lint_javascript` in the CI) has a lot of suggestions that will require manual changes; we won't make it a required test in this PR.

### CODE REVIEW

**Comment by skearnes** [comment]
Moving this to draft; we can check it in after #242 is done.

**Comment by skearnes** [comment]
@antonkast-google @n8kim1 I think clang-format broke something; can you help me find it?

https://github.com/Open-Reaction-Database/ord-schema/pull/246/checks?check_run_id=951646157

```shell
node js/test.js
127.0.0.1 - - [06/Aug/2020 00:12:27] "GET /dataset/empty/reaction/0?user=test HTTP/1.1" 302 -
127.0.0.1 - - [06/Aug/2020 00:12:27] "GET /dataset/empty/reaction/0?user=test HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:27] "GET /js/reaction.js HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:27] "GET /css/reaction.css HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:27] "GET /css/ketcher-bootstrap.css HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:28] "GET /ketcher/iframe HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:28] "GET /ketcher/ketcher.css HTTP/1.1" 200 -
127.0.0.1 - - [06/Aug/2020 00:12:28] "GET /ketcher/ketcher.js HTTP/1.1" 200 -
```

```js
(node:4466) UnhandledPromiseRejectionWarning: TimeoutError: waiting for selector "body[ready=true]" failed: timeout 30000ms exceeded
    at new WaitTask (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:394:34)
    at DOMWorld._waitForSelectorOrXPath (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:326:26)
    at DOMWorld.waitForSelector (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:309:21)
    at Frame.waitForSelector (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:801:51)
    at Frame.waitFor (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:757:25)
    at Page.waitFor (/home/runner/work/ord-schema/ord-schema/editor/node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js:1212:33)
    at /home/runner/work/ord-schema/ord-schema/editor/js/test.js:41:16
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:4466) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:4466) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by n8kim1** [comment]
The linter broke reaction.js in a couple places... After changing those manually, the tests pass. I'll see if I can format them in a way that makes the linter happy.

I'm also worried that the linter may have broken other parts of the JS that we haven't found.

**Comment by n8kim1** [comment]
I noticed a missing newline at the end of notes.js. Could I add it here (since we're making all sorts of formatting fixes anyways), or should I leave it for another PR?

**Comment by skearnes** [comment]
> I noticed a missing newline at the end of notes.js. Could I add it here (since we're making all sorts of formatting fixes anyways), or should I leave it for another PR?

Yes, that's fine.

**Comment by skearnes** [comment]
It looks like clang-format correctly found places where we forgot semicolons and merged these lines; these are good things to catch and I don't see any errors actually introduced by the formatter (just revealed).

File: ord_schema/validations_test.py [hunk 1]
```diff
@@ -207,9 +203,8 @@ def test_reaction_recursive(self):
         # Assigning internal standard role to input should resolve the error
         message_input_istd = reaction_pb2.Reaction()
         message_input_istd.CopyFrom(message)
-        message_input_istd.inputs['dummy_input'].components[
```
Comment (connorcoley) [inline ord_schema/validations_test.py:210]: I can't imagine a human decided to use these line breaks, so just a note that it seems like we've changed what automatic styling is enforced

Comment (skearnes) [reply ord_schema/validations_test.py:210]: Yes, we're using the "Google" styling now; I prefer how it handles e.g. dicts compared to before.

File: ord_schema/proto/reaction.proto [hunk 2]
```diff
@@ -274,10 +274,10 @@ message CompoundPreparation {
   enum PreparationType {
     UNSPECIFIED = 0;
     CUSTOM = 1;
-    NONE = 2;  // Compound used as received.
+    NONE = 2;        // Compound used as received.
```
[the remark this answers is not in the document]
Comment (skearnes) [reply ord_schema/proto/reaction.proto:277]: In general I like the consistent visualization for inline comments within a message. It looks like this particular instance didn't behave well (compared to AnalysisType below, for example). I'll fix that manually but I'd like to keep using clang-format here to enforce some consistency programmatically if that's ok.

[the remark this answers is not in the document]
Comment (skearnes) [reply ord_schema/proto/reaction.proto:277]: Huh, it reverted my changes; I'll see if there's a config option for that.

File: ord_schema/validations_test.py (hunk 1 above)
Comment (skearnes) [reply ord_schema/validations_test.py:210]: Updated .style.yapf with these style options so we have more control over this in future.

File: ord_schema/proto/reaction.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (skearnes) [reply ord_schema/proto/reaction.proto:277]: I found the relevant param and disabled it in the newly added `.clang-format`; now all comments will be two spaces.

**Pull request merged**